Sources/SwiftKuery/ConnectionPoolConnection.swift

```diff
@@ -16,12 +16,27 @@
 
 // MARK: ConnectionPoolConnection
 
-/// A wrapper around actual `Connection`. Allows seemless release of the connection.
+/**
+ This class uses a `Connection` instance and a `ConnectionPool` instance to implements a wrapper for the `Connection` class.
+ It implements the functions of the `Connection` protocol, in addition to the `closeConnection()` function
+ for releasing a `Connection` instance from the `ConnectionPool`.
+ It is usually initialised by the `ConnectionPool` instance.
+ ### Usage Example: ###
+ In this example, a `ConnectionPool` instance is initialized (parameters defined in docs for `ConnectionPool`).
+ A `ConnectionPoolConnection` instance called "connection" is then returned by calling "getConnection()" on "connectionPool".
```

Comment:
To cut the wording a little how about "A `ConnectionPoolConnection` instance is created by calling `connectionPool.getConnection()`. This connection is then released from the pool."